analyzer/src/main/kotlin/managers/Yarn2.kt

```diff
@@ -63,20 +93,56 @@ class Yarn2(
      */
     private val yarn2ExecutablesByPath: MutableMap<File, String> = mutableMapOf()
 
-    /**
-     * A NPM package manager, used to call NPM when Yarn 2+ does not provide enough information.
-     */
-    private val delegateNpmAnalyzer = Npm("npm", analysisRoot, analyzerConfig, repoConfig)
+    private val disableRegistryCertificateVerification = analyzerConfig
+        .getPackageManagerConfiguration(managerName)?.options
+        ?.get(OPTION_DISABLE_REGISTRY_CERTIFICATE_VERIFICATION).toBoolean()
 
     companion object {
         // The name of Yarn 2+ resource file.
         const val YARN2_RESOURCE_FILE = ".yarnrc.yml"
 
         // The property in `.yarnrc.yml`containing the path to the Yarn2+ executable.
         const val YARN_PATH_PROPERTY_NAME = "yarnPath"
+
+        // The name of the scope with the regular dependencies.
+        private enum class DependencyType(val type: String) {
+            DEPENDENCIES("dependencies"),
+            DEV_DEPENDENCIES("devDependencies")
+        }
```

Comment:
I guess there's no reason to have this `enum class` inside a (static) companion object. It should be moved to the top level, like the other private properties, too.